src/Repository/SettingRepository.php

```
@@ -0,0 +1,18 @@
+<?php
+
+namespace Letsgoi\LaravelSettings\Repository;
+
+use Illuminate\Support\Facades\Cache;
+use Letsgoi\LaravelSettings\Models\Setting;
+
+class SettingRepository
+{
+    public function find(string $key): string|float|array|bool|null
+    {
+        return Cache::rememberForever($key, static function () use ($key) {
+            $setting = Setting::firstWhere('key', $key);
+
+            return $setting?->value;
```

Maybe it can throw an exception when the setting does not exist, instead of returning null.

---

done